src/util/parented_ptr.h

```diff
@@ -0,0 +1,78 @@
+#ifndef PARENTED_POINTER_H
+#define PARENTED_POINTER_H
+
+#include <memory>
+#include "assert.h"
+
+
+/**
+ * Use this wrapper class to clearly represent a pointer that is owned by the QT object tree.
+ * Objects which both derive from QObject AND have a parent object, have their lifetime governed by the QT object tree, 
+ * and thus pointers to such objects do not need to be deleted when they go  out of scope.
+**/
+template <typename T>
+class parented_ptr {
+  public:
+    explicit parented_ptr(T* t) : m_pPtr(t) {
+        // use C style assert here as it usefully prints the type T
+        assert(t->parent() != nullptr);
+    }
+
+    parented_ptr() : m_pPtr(nullptr) {}
+
+    ~parented_ptr() = default;
+
+    T* get() const {
+        return m_pPtr;
```

Comment:
In case of shared and uique pointer get() indicates a pointer usage without passing ownership. 
If we use get() here as well, it looks similar, but the whole new pointer does not have the ownership, so I think we can remove get and add a cast operator instead. This way we can get rid of the get cat() calls at the user side.